dingo/experimental_gurobi_functions/redundant_facet_removal_experiment.py

```diff
@@ -0,0 +1,300 @@
+# dingo : a python library for metabolic networks sampling and analysis
+# dingo is part of GeomScale project
+
+# Copyright (c) 2021 Apostolos Chalkis
+
+# Licensed under GNU LGPL.3, see LICENCE file
+
+import sys
+import numpy as np
+import scipy.sparse as sp
+import gurobipy as gp
+from gurobipy import GRB
+import math
+
+def update_model_constraints_and_bounds(model, Aeq_sparse=None, beq=None, A_sparse=None, b=None,
+                                       new_lower_bounds=None, new_upper_bounds=None):
+    if Aeq_sparse is not None and beq is not None:
+        # Remove the old equality constraints
+        model.remove("c")
+
+        # Add new equality constraints
+        model.addMConstr(sp.csr_matrix(Aeq_sparse), model.getVars(), "=", np.array(beq), name="c")
+
+    if A_sparse is not None and b is not None:
+        # Remove the old inequality constraints
+        model.remove("d")
+
+        # Add new inequality constraints
+        model.addMConstr(sp.csr_matrix(A_sparse), model.getVars(), "<", np.array(b), name="d")
+
+    if new_lower_bounds is not None or new_upper_bounds is not None:
+        # Retrieve the variable objects
+        x_vars = model.getVars()
+
+        # Update variable bounds if specified
+        if new_lower_bounds is not None:
+            for i, x_var in enumerate(x_vars):
+                x_var.lb = new_lower_bounds[i]
+
+        if new_upper_bounds is not None:
+            for i, x_var in enumerate(x_vars):
+                x_var.ub = new_upper_bounds[i]
+
+    # Update the model to apply the changes
+    model.update()
+
+# Usage example:
+# Assuming you have a Gurobi model 'model' and new constraint matrices/vectors and bounds
+# Call the function to update the model
+# update_model_constraints_and_bounds(model, Aeq_sparse_new, beq_new, A_sparse_new, b_new, new_lower_bounds, new_upper_bounds)
+
+
+def solve_lp_with_different_objectives(model, new_objective_coeffs):
+    """
+    Solve a linear program with a different objective function.
+
+    Parameters:
+        model (gurobipy.Model): The Gurobi model with the original constraints.
+        new_objective_coeffs (list): List of new objective coefficients for the variables.
+
+    Returns:
+        gurobipy.Model: The updated Gurobi model with the new objective function.
+    """
+    # Clear the existing objective function
+    model.setObjective(0, clear=True)
+
+    # Update the objective function with the new coefficients
+    for i, var in enumerate(model.getVars()):
```

Comment:
optimization todo: you could pass a numpy array here (instead of a list of coefficients) and update the objective in one call without a for loop.

It is ok if you do not implement this in this PR but please add it as a comment

Reply:
updated it